frontendservice/src/views/SignUp.vue

```diff
@@ -0,0 +1,172 @@
+/* Copyright 2020 Google LLC Licensed under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance with the License. You
+may obtain a copy of the License at https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License. */
+
+<template>
+  <div class="sign-up">
+    <h1>Sign Up</h1>
+    <span>New to Virtual Queues? Create an account now!</span>
+    <div class="input-icons">
+      <i class="fa fa-user icon"></i>
+      <input class="input-field" type="text" v-model="name" placeholder="Name" />
+    </div>
+    <p>{{nameAlert}}</p>
+    <div class="input-icons">
+      <i class="fa fa-envelope icon"></i>
+      <input class="input-field" type="text" v-model="email" placeholder="Email" />
+    </div>
+    <p>{{emailAlert}}</p>
+    <div class="input-icons">
+      <i @click="toggle" :class="{'fa fa-eye icon': isHidden, 'fa fa-eye-slash icon': !isHidden}"></i>
+      <input class="input-field" ref="password" type="password" v-model="password" placeholder="Password" />
+    </div>
+    <p>{{passwordAlert}}</p>
+    <a class="button is-primary" @click="createUser">Sign Up</a>
+    <p>{{firebaseAlert}}</p>
+    <div>
+      If you have already registered, go to <router-link to="/login">login</router-link>.
+    </div>
+    <div>
+      Or go back to <router-link to="/about">about</router-link>.
+    </div>
+  </div>
+</template>
+
+<script>
+import firebase from "firebase";
+export default {
+  name: "signUp",
+  data() {
+    return {
+      email: "",
+      password: "",
+      name: "",
+      isHidden: true,
+      nameAlert: "",
+      emailAlert: "",
+      passwordAlert: "",
+      firebaseAlert: ""
+    };
+  },
+  methods: {
+    toggle() {
+      var passwordRef = this.$refs.password;
+      if (passwordRef.type === "password") {
+        passwordRef.type = "text";
+      } else {
+        passwordRef.type = "password";
+      }
+      this.isHidden = !this.isHidden;
+    },
+    createUser: function() {
+      this.nameAlert = "";
+      this.emailAlert = "";
+      this.passwordAlert = "";
+      this.firebaseAlert = "";
+
+      //to check if the input email follows correct format like *@*.*
```

Comment:
Add a space between "//" and the first letter of the comment. Similarly for others

Reply:
Added.